src/compiler/types.ts

```diff
@@ -2098,6 +2138,7 @@ namespace ts {
 
     export interface JSDoc extends Node {
         kind: SyntaxKind.JSDocComment;
+        parent?: HasJSDoc;

```

Comment:
`& Node` ?

Reply:
All members of `HasJSDoc` are already nodes. The interface `HasJSDocNodes` would require that. Is that confusing? Should I rename the second to something like `JSDocContainer`?

Reply:
JSDocContainer seems like a better name